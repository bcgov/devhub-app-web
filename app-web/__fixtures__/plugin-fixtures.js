/*
Copyright 2018 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/
export const SOURCE_DEVHUB_GITHUB_QL_NODE = {
  id: 'd90c1b12-9055-580b-adb3-b29354871657',
  source: 'design-system',
  sourceName: 'Design System',
  sourcePath: 'https://github.com/bcgov/design-system/',
  resourcePath: '/design-system/README_123o8123',
  path: 'components/footer/README.md',
  owner: 'bcgov',
  labels: ['Design System', 'Components'],
  fileName: 'about.md',
  fileType: 'Markdown',
  pagePath: '/design-system/about_G0Z6oO3_3',
  originalSource:
    'https://github.com/bcgov/design-system/blob/master/components/footer/something/README.md',
  internal: {
    mediaType: 'text/markdown',
    type: 'SourceDevhubGithub',
    owner: 'gatsby-source-github-all',
    contentDigest: '1239087asdfjkas',
    content: '---\nsomething:true\n---\n# Hello world\n',
  },
  childMarkdownRemark: {
    id: 'c03ba7af-a099-5dbe-8aec-5922922e0fcc >>> MarkdownRemark',
    html: '<h1>Hello World</h1>',
    frontmatter: {
      something: 'true',
      _PARENT: 'c03ba7af-a099-5dbe-8aec-5922922e0fcc',
      parent: 'c03ba7af-a099-5dbe-8aec-5922922e0fcc',
      status: null,
    },
    headings: [
      {
        value: 'Hello World',
        depth: 1,
      },
    ],
    internal: {
      content:
        '---\ndescription: What the Design System is and how it works.\ntitle: About\n---\n\n## About\n\nThe Design System helps developers and designers build better digital products and services.\n\nIt’s a collection of digital resources and tools – including a library of reusable UI/interface components and service patterns. The system makes it easier and faster to build custom B.C. government websites and applications. \n\nComponents are collectively built by the government community, meet accessibility standards and are open for input and improvement.\n\nUse the Design System if you’re building custom B.C. government and applications. For more information about applying for external URLs, contact the B.C. government Joint Working Group.\n\n## The Team\n\nThe Design System is managed by a team of designers and developers and the government community. The system is a part of the DevHub, a project by the Developers’ Exchange that brings together UI/interface and back-end resources for digital teams.\n\nKelsey Singbeil – Team Lead, Service Design\n\nBryan Smith – Service Design Specialist\n\nOri Nevares – Service Design Specialist\n\n## The Community\n\nThe system is collectively built and maintained by the government community. It’s open source, which means that anybody can provide input, offer suggestions and get involved.\n\nShare your feedback to make this resource better for everyone.\n\n## How the system works\n\nHow might we improve people’s interaction with B.C. government digital products and services?\n\nThe system is a part of the answer.\n\nIt provides developers and designers with the tools they need to build great digital government products and services. The system is collectively built and managed, so all the tools and resources are the product of the community.\n\nAs part of this, we are committed to making all the tools and resources we create:\n\n**Maintained**  \nThe community and Design System management team will support the maintenance and development of new resources and tools. This means you can flag issues and they’ll be addressed, discuss rationale and directly chat with the management team whenever you need.\n\n**Collaborative**\nThe system is only valuable if the community finds it valuable. If a tool or resource isn’t meeting your needs, tell us why and work with us to make it better. We’ll also come to you – in your office, at your desk and share what’s new, ask what you need and share how you can get involved.\n\n**Open**\nThe system is built in GitHub and meant to be shared and improved by everyone. The code is open source and our work is in the open. We are transparent about how choices are made and tools are improved. Ask us…anything.\n\n**Contribute to the Design System**\n\n***Help with design***\n-   Suggest a component\n-   Improve documentation\n\n***Help with code***\n-  Code contribution guide\n-   Report an issue\n\n***Help with usability***\n-   Component discussions\n-   Provide research findings\n\n',
      type: 'MarkdownRemark',
      contentDigest: 'b186249c65e12e08344e31e289d99ab8',
      owner: 'gatsby-transformer-remark',
    },
  },
};

export const FILE_QL_NODE = {
  id: '/foo/Index.test.js absPath of file',
  internal: {
    contentDigest: '2a9e4d519252bf9f9ccc9a931d2d731e',
    mediaType: 'application/javascript',
    type: 'File',
    description: 'File "src/pages/Index.test.js"',
    owner: 'gatsby-source-filesystem',
  },
  birthtime: '2018-10-24T21:16:43.521Z',
  publicURL: '/static/Index.test-2a9e4d519252bf9f9ccc9a931d2d731e.js',
  ctime: '2018-10-24T21:16:43.522Z',
  mtime: '2018-10-24T21:16:43.522Z',
  atime: '2018-10-25T20:32:50.743Z',
  birthtimeMs: 1540415803521.444,
  ctimeMs: 1540415803521.575,
  mtimeMs: 1540415803521.575,
  atimeMs: 1540499570743.2075,
  sourceInstanceName: 'pages',
  absolutePath: '/foo/Index.test.js',
  relativePath: 'Index.test.js',
  relativeDirectory: '',
  extension: 'js',
  size: 1674,
  prettySize: '1.67 kB',
  name: 'Index.test',
  ext: '.js',
  base: 'Index.test.js',
  dev: 16777220,
  mode: 33188,
  nlink: 1,
  uid: 323756479,
  gid: 1839645156,
  blocks: 8,
  ino: 11739371,
  blksize: 4194304,
  rdev: 0,
};
