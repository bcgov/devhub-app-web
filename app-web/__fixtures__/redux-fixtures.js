/*
Copyright 2018 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/
import * as actions from '../src/store/actions/actionTypes';
export const ACTIONS = {
  LOAD_SIPHON_NODES: {
    type: actions.LOAD_SIPHON_NODES,
    payload: {
      nodes: [{ foo: 'bar' }, { foo: 'baz' }, { foo: 'foo' }],
    },
  },
  FILTER_SIPHON_NODES: {
    type: actions.FILTER_SIPHON_NODES,
    payload: {
      filteredBy: 'foo',
      value: 'bar',
    },
  },
  FILTER_SIPHON_NODES_BY_ALL: {
    type: actions.FILTER_SIPHON_NODES,
    payload: {
      filteredBy: 'foo',
      value: 'All',
    },
  },
  SELECT_FILTER_OPTION: {
    type: actions.SET_SELECTED_FILTER_OPTION,
    payload: {
      foo: 'bar',
    },
  },
  TOGGLE_MAIN_NAVIGATION_ON: {
    type: actions.TOGGLE_MAIN_NAVIGATION,
    payload: {
      toggled: true,
    },
  },
  TOGGLE_MAIN_NAVIGATION_OFF: {
    type: actions.TOGGLE_MAIN_NAVIGATION,
    payload: {
      toggled: false,
    },
  },
};

export const INITIAL_STATES = {
  SIPHON: {
    nodes: [],
    filteredNodes: [],
    groupBy: null,
    loading: false,
    error: false,
    messages: [],
  },
  UI: {
    selectedFilterOption: null,
    mainNavigationToggled: false,
  },
};
