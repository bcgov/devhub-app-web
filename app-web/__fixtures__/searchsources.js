/*
Copyright 2019 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/
// run time apollo query fixtures

export const ROCKET_CHAT = [
  {
    id: '1', 
    roomId: '2', author: 'apple', time: '2019-02-12', message: 'apples are clearly superior',
  },
  {
    id: '2', 
    roomId: '2', author: 'orange', time: '2019-02-12', message: 'Nice joke apple, are trying to perform some time of standup?',
  },
  {
    id: '3', 
    roomId: '2', author: 'apple', time: '2019-02-12', message: 'Don\'t play coy with me orange, you know I\'m telling the truth',
  },
  {
    id: '4', 
    roomId: '2', author: 'orange', time: '2019-02-12', message: 'I ain\'t never hurd of people curing scurvy with apples yo. Hook line and sinker dawg.',
  },
]