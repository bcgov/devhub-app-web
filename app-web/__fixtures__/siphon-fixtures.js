import { arrayToMapByProp } from '../src/utils/dataHelpers';
import { RESOURCE_TYPES } from '../src/constants/ui';

const IDS = {
  DESIGN_SYSTEM: 'collection-1',
  DEVHUB: 'collection-2',
};

// SIPHON NODES
export const DESIGN_SYSTEM_NODE_1 = {
  id: '1',
  parent: {
    id: IDS.DESIGN_SYSTEM,
  },
  _metadata: {
    position: '000.000.000',
  },
  attributes: {
    personas: ['Designer'],
  },
  source: {
    displayName: 'Design System',
  },
  resource: {
    path: '/design-system/Z1bGIqu',
    type: 'Components',
  },
  unfurl: {
    type: 'markdown',
    label1: '',
    data1: '',
    label2: '',
    data2: '',
    image: 'https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true',
    title: 'About',
    description: 'What the Design System is and how it works.',
    author: 'ksingbeil',
  },
  childMarkdownRemark: {
    frontmatter: {
      pageOnly: false,
    },
  },
};

export const DESIGN_SYSTEM_NODE_2 = {
  id: '2',
  parent: {
    id: IDS.DESIGN_SYSTEM,
  },
  _metadata: {
    position: '000.000.001',
  },
  attributes: {
    personas: ['Designer'],
  },
  source: {
    displayName: 'Design System',
  },
  resource: {
    path: '/design-system/Z1bGIqu',
    type: 'Components',
  },
  unfurl: {
    type: 'markdown',
    label1: '',
    data1: '',
    label2: '',
    data2: '',
    image: 'https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true',
    title: 'About',
    description: 'What the Design System is and how it works.',
    author: 'ksingbeil',
  },
  childMarkdownRemark: {
    frontmatter: {
      pageOnly: false,
    },
  },
};

export const DESIGN_SYSTEM_NODE_3 = {
  id: '3',
  parent: {
    id: IDS.DESIGN_SYSTEM,
  },
  _metadata: {
    position: '000.000.002',
  },
  attributes: {
    personas: ['Designer'],
  },
  source: {
    displayName: 'Design System',
  },
  resource: {
    path: '/design-system/Z1bGIqu',
    type: 'Components',
  },
  unfurl: {
    type: 'markdown',
    label1: '',
    data1: '',
    label2: '',
    data2: '',
    image: 'https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true',
    title: 'About',
    description: 'What the Design System is and how it works.',
    author: 'ksingbeil',
  },
  childMarkdownRemark: {
    frontmatter: {
      pageOnly: false,
    },
  },
};

export const DESIGN_SYSTEM_NODE_4 = {
  id: '4',
  parent: {
    id: IDS.DESIGN_SYSTEM,
  },
  _metadata: {
    position: '000.000.003',
  },
  attributes: {
    personas: ['Developer'],
  },
  source: {
    displayName: 'Design System',
  },
  resource: {
    path: '/design-system/Z1bGIqu',
    type: 'Components',
  },
  unfurl: {
    type: 'markdown',
    label1: '',
    data1: '',
    label2: '',
    data2: '',
    image: 'https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true',
    title: 'About',
    description: 'What the Design System is and how it works.',
    author: 'ksingbeil',
  },
  childMarkdownRemark: {
    frontmatter: {
      pageOnly: false,
    },
  },
};

export const DEVHUB_NODE_1 = {
  id: '5',
  parent: {
    id: IDS.DEVHUB,
  },
  _metadata: {
    position: '001.000.000',
  },
  attributes: {
    personas: ['Developer'],
  },
  source: {
    displayName: 'Design System',
  },
  resource: {
    path: '/design-system/Z1bGIqu',
    type: 'Documentation',
  },
  unfurl: {
    type: 'markdown',
    label1: '',
    data1: '',
    label2: '',
    data2: '',
    image: 'https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true',
    title: 'About',
    description: 'What the Design System is and how it works.',
    author: 'ksingbeil',
  },
  childMarkdownRemark: {
    frontmatter: {
      pageOnly: false,
    },
  },
};

export const DEVHUB_NODE_2 = {
  id: '6',
  parent: {
    id: IDS.DEVHUB,
  },
  _metadata: {
    position: '001.000.001',
  },
  attributes: {
    personas: ['Developer'],
  },
  source: {
    displayName: 'Design System',
  },
  resource: {
    path: '/design-system/Z1bGIqu',
    type: 'Documentation',
  },
  unfurl: {
    type: 'markdown',
    label1: '',
    data1: '',
    label2: '',
    data2: '',
    image: 'https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true',
    title: 'About',
    description: 'What the Design System is and how it works.',
    author: 'ksingbeil',
  },
  childMarkdownRemark: {
    frontmatter: {
      pageOnly: false,
    },
  },
};

export const DEVHUB_NODE_3 = {
  id: '7',
  parent: {
    id: IDS.DEVHUB,
  },
  _metadata: {
    position: '001.000.001',
  },
  attributes: {
    personas: ['Product Owner'],
  },
  source: {
    displayName: 'Design System',
  },
  resource: {
    path: '/design-system/Z1bGIqu',
    type: 'Documentation',
  },
  unfurl: {
    type: 'markdown',
    label1: '',
    data1: '',
    label2: '',
    data2: '',
    image: 'https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true',
    title: 'About',
    description: 'What the Design System is and how it works.',
    author: 'ksingbeil',
  },
  childMarkdownRemark: {
    frontmatter: {
      pageOnly: false,
    },
  },
};

export const DESIGN_SYSTEM_NODES = [
  DESIGN_SYSTEM_NODE_3,
  DESIGN_SYSTEM_NODE_2,
  DESIGN_SYSTEM_NODE_1,
  DESIGN_SYSTEM_NODE_4,
];

export const DEVHUB_NODES = [DEVHUB_NODE_1, DEVHUB_NODE_2, DEVHUB_NODE_3];

// COLLECTIONS
export const DESIGN_SYSTEM_COLLECTION = {
  id: IDS.DESIGN_SYSTEM,
  type: 'default',
  title: 'Design System',
  description: 'baz',
  resources: DESIGN_SYSTEM_NODES.map(n => ({ id: n.id })),
};

export const DEVHUB_COLLECTION = {
  id: IDS.DEVHUB,
  type: 'default',
  title: 'Devhub',
  description: 'baz',
  resources: DEVHUB_NODES.map(n => ({ id: n.id })),
};

export const SIPHON_NODES = DESIGN_SYSTEM_NODES.concat(DEVHUB_NODES);

// stubbing in a set of nodes that pass the designer filter
export const FILTERED_NODES = SIPHON_NODES.filter(
  node => node.attributes.personas[0] === 'Designer',
);

export const COLLECTIONS = [DESIGN_SYSTEM_COLLECTION, DEVHUB_COLLECTION];

export const SIPHON_NODES_MAP = arrayToMapByProp(SIPHON_NODES, 'id');

export const COLLECTIONS_MAP = arrayToMapByProp(COLLECTIONS, 'id');

// some selectors found within /selectors/index.js provide selections of resources grouped by
// resources types they are combined wth the default grouping set so that all resources
// have a map of all resource types like so
/**
 *
 * {
 *  Components: [] no resources so it defaulted to empty,
 *  Documentation: [{}, {}],
 *  ...etc
 * }
 */
export const DEFAULT_GROUPINGS = Object.keys(RESOURCE_TYPES).reduce((grouping, type) => {
  grouping[RESOURCE_TYPES[type]] = [];
  return grouping;
}, {});
