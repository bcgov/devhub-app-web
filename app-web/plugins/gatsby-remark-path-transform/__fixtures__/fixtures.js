const PARAGRAPH_AST = {
    'type': 'paragraph',
    'children': [
      {'type': 'text', 'value': 'GITHUB-EMBED GITHUB-EMBED'}
    ],
  };

const IMAGE_AST_ABSOLUTE = {
    'type': 'image',
    'url': 'https://www.google.com/images/dog.png',
    'children': [
      {'type': 'text', 'value': 'GITHUB-EMBED GITHUB-EMBED'}
    ],
};

const IMAGE_AST_RELATIVE = {
    'type': 'image',
    'url': '../images/dog.png',
    'children': [
      {'type': 'text', 'value': 'GITHUB-EMBED GITHUB-EMBED'}
    ],
};

const LINK_AST_ABSOLUTE = {
    'type': 'link',
    'url': 'https://www.google.com/images/dog.png',
    'children': [
      {'type': 'text', 'value': 'GITHUB-EMBED GITHUB-EMBED'}
    ],
};

const LINK_AST_RELATIVE = {
    'type': 'link',
    'url': '../images/dog.png',
    'children': [
      {'type': 'text', 'value': 'GITHUB-EMBED GITHUB-EMBED'}
    ],
};

const GRAPH_QL_PARENT_NODE = {
    internal: {
        type: 'node',
        mediaType: 'text',
    },
    parent: null,
};
const MARKDOWN_NODE = {
    internal: {
        type: 'node',
        mediaType: 'text',
    },
    parent: null,
};
module.exports = {
    PARAGRAPH_AST,
    IMAGE_AST_ABSOLUTE,
    IMAGE_AST_RELATIVE,
    LINK_AST_ABSOLUTE,
    LINK_AST_RELATIVE,
    GRAPH_QL_PARENT_NODE,
    MARKDOWN_NODE,
};
