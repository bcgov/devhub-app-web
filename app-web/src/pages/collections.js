/*
Copyright 2019 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/
import React, { Component } from 'react';

import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import * as actions from '../store/actions';
import * as selectors from '../store/selectors';

import { COLLECTIONS_PAGE } from '../messages';
import { flattenGatsbyGraphQL } from '../utils//dataHelpers';

import Title from '../components/Page/Title';
import CollectionPreview from '../components/CollectionPreview/CollectionPreview';
import Main from '../components/Page/Main';
import withResourceQuery from '../hoc/withResourceQuery';
import Layout from '../hoc/Layout';
import { getFirstNonExternalResource } from '../utils/helpers';

export class CollectionsPage extends Component {
  componentDidMount() {
    // flatted nodes from graphql
    if (!this.props.resourcesLoaded) {
      const collections = flattenGatsbyGraphQL(this.props.data.allDevhubSiphonCollection.edges);
      // note this.props.data is received from the withResourceQuery Component
      const resources = flattenGatsbyGraphQL(this.props.data.allDevhubSiphon.edges);
      this.props.loadResources(resources, collections);
    }
  }

  render() {
    const { collections } = this.props;
    return (
      <Layout>
        <Main>
          <Title
            title={COLLECTIONS_PAGE.header.title.defaultMessage}
            subtitle={COLLECTIONS_PAGE.header.subtitle.defaultMessage}
          />
          {collections.map(collection => (
            <CollectionPreview
              key={collection.id}
              title={collection.name}
              description={collection.description}
              resources={collection.resources}
              link={getFirstNonExternalResource(collection.resources)}
            />
          ))}
        </Main>
      </Layout>
    );
  }
}

const mapStateToProps = createStructuredSelector({
  resourcesLoaded: selectors.selectResourcesLoaded,
  collections: selectors.selectCollectionsWithResources,
});

const mapDispatchToProps = dispatch => {
  return {
    loadResources: (resources, collections) =>
      dispatch(actions.loadResources(resources, collections)),
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(withResourceQuery(CollectionsPage)());
