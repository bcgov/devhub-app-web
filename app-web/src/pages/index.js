import React, { PureComponent } from 'react';
import { createStructuredSelector } from 'reselect';
import queryString from 'query-string';
import shortid from 'shortid';
import { connect } from 'react-redux';
import { REACT_SCROLL } from '../constants/ui';
import { MAIN_NAV_ROUTES } from '../constants/routes';
import { flattenGatsbyGraphQL } from '../utils/dataHelpers';
import { getSearchResults } from '../utils/helpers';
import * as actions from '../store/actions';

import styles from './index.module.css';
// components
import { Flag } from 'flag';
import { Element } from 'react-scroll';
import Loading from '../components/UI/Loading/Loading';
import Layout from '../hoc/Layout';
import ResourcePreview from '../components/ResourcePreview/ResourcePreview';
import Masthead from '../components/Home/Masthead';

// selectors from reselect
import {
  selectQuery,
  selectSearchResultsLength,
  selectTotalResources,
  selectSearchWordLength,
  selectResourcesLoaded,
  selectResourcesReducerLoading,
  selectGroupedFilteredAvailableResources,
  selectCollectionsWithAvailableResourcesGroupedByType,
} from '../store/selectors';

import { SEARCH } from '../messages';
import withResourceQuery from '../hoc/withResourceQuery';
import CollectionsContainer from '../components/Page/CollectionsContainer';
import Aux from '../hoc/auxillary';

export class Index extends PureComponent {
  componentDidMount() {
    // flatted nodes from graphql
    if (!this.props.resourcesLoaded) {
      const collections = flattenGatsbyGraphQL(this.props.data.allDevhubSiphonCollection.edges);
      // note this.props.data is received from the withResourceQuery Component
      const resources = flattenGatsbyGraphQL(this.props.data.allDevhubSiphon.edges);
      this.props.loadResources(resources, collections);
    }
    // reset resource type to null since index page views all index pages
    this.props.setResourceType(null);
  }

  componentDidUpdate() {
    const query = queryString.parse(this.props.location.search);
    if (Object.prototype.hasOwnProperty.call(query, 'q')) {
      const param = decodeURIComponent(query.q);

      if (param !== this.props.query) {
        this.props.setSearchQuery(param);
        this.getSearchResults(param).then(results => {
          this.props.setSearchResults(results);
        });
      }
    }
  }

  componentWillUnmount() {
    // unset set all search properties so that when this page is navigated back to, it looks like a fresh
    // page
    this.props.resetSearch();
  }

  /**
   * gets search results from lunr
   * @param {String} query the search string
   */
  async getSearchResults(query) {
    const lunr = await window.__LUNR__.__loaded;
    return getSearchResults(query, lunr);
  }

  render() {
    const {
      resourcesByType,
      searchResultsLength,
      setSearchBarTerms,
      searchWordLength,
      collections,
    } = this.props;

    const SiphonResources = Object.keys(resourcesByType).map(resourceType => {
      if (resourcesByType[resourceType].length > 0) {
        return (
          <ResourcePreview
            key={shortid.generate()}
            title={resourceType}
            resources={resourcesByType[resourceType]}
            link={MAIN_NAV_ROUTES[resourceType]}
          />
        );
      }
      return null;
    });

    return (
      <Layout showHamburger>
        <div>
          <Masthead setSearchBarTerms={setSearchBarTerms} />
          <main role="main" className={styles.Main}>
            {this.props.loading ? (
              <Loading message="Loading..." />
            ) : searchResultsLength === 0 && searchWordLength > 0 ? (
              <p>{SEARCH.results.empty.defaultMessage}</p>
            ) : (
              <Aux>
                <CollectionsContainer collections={collections} />
                <Element name={REACT_SCROLL.ELEMENTS.CARDS_CONTAINER}>
                  {/* Element used for react-scroll targeting */}
                  <Flag name="features.githubResourceCards">{SiphonResources}</Flag>
                </Element>
              </Aux>
            )}
          </main>
        </div>
      </Layout>
    );
  }
}

const mapStateToProps = createStructuredSelector({
  resourcesLoaded: selectResourcesLoaded,
  query: selectQuery,
  loading: selectResourcesReducerLoading,
  searchResultsLength: selectSearchResultsLength,
  totalResources: selectTotalResources,
  searchWordLength: selectSearchWordLength,
  resourcesByType: selectGroupedFilteredAvailableResources,
  collections: selectCollectionsWithAvailableResourcesGroupedByType,
});

const mapDispatchToProps = dispatch => {
  return {
    loadResources: (resources, collections) =>
      dispatch(actions.loadResources(resources, collections)),
    setSearchResults: results => dispatch(actions.setSearchResults(results)),
    setSearchQuery: query => dispatch(actions.setSearchQuery(query)),
    setSearchBarTerms: resourceType => dispatch(actions.setSearchBarTerms(resourceType)),
    resetSearch: () => dispatch(actions.resetSearch()),
    setResourceType: type => dispatch(actions.setResourceType(type)),
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(withResourceQuery(Index)());
