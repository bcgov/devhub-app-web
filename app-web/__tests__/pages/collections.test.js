import React from 'react';
import { shallow } from 'enzyme';
import { CollectionsPage } from '../../src/pages/collections';
import { SIPHON_NODES, COLLECTIONS } from '../../__fixtures__/siphon-fixtures';
jest.mock('react-spinners', () => null);

describe('Index Container', () => {
  test('it matches snapshot', () => {
    // when you use graphql to load data into the component
    // all edges are an object of { node: [graphql object]}
    // the collections fixture is the true data without this extra object field
    // so we map it to resemble what graphql would do when passing the data attribute into
    // this component
    const nodes = SIPHON_NODES.map(c => ({ node: c }));
    const collections = COLLECTIONS.map(c => ({ node: c }));
    const data = {
      allDevhubSiphon: {
        edges: nodes,
      },
      allDevhubSiphonCollection: {
        edges: collections,
      },
    };
    // mocking redux actions
    const actions = {
      loadResources: jest.fn(),
      setSearchResults: jest.fn(),
      setSearchQuery: jest.fn(),
      setSearchBarTerms: jest.fn(),
      resetSearch: jest.fn(),
      setResourceType: jest.fn(),
    };
    const props = {
      data,
      collections: COLLECTIONS.map(c => ({
        ...c,
        resources: c.resources.map(r => ({ ...r, resource: { path: '/' } })),
      })),
      location: {
        pathname: '/collections',
      },
      resourcesLoaded: false,
    };

    const wrapper = shallow(<CollectionsPage {...props} {...actions} />);
    expect(wrapper).toMatchSnapshot();
  });
});
