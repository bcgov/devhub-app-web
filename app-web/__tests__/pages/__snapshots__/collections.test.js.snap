// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Container it matches snapshot 1`] = `
<Connect(Layout)>
  <Main>
    <Title
      subtitle="Explore resources that have been specially curated to enhance context, detail and accessibility."
      title="Collections"
    />
    <CollectionPreview
      description="baz"
      key="collection-1"
      link={null}
      resources={
        Array [
          Object {
            "id": "3",
            "resource": Object {
              "path": "/",
            },
          },
          Object {
            "id": "2",
            "resource": Object {
              "path": "/",
            },
          },
          Object {
            "id": "1",
            "resource": Object {
              "path": "/",
            },
          },
          Object {
            "id": "4",
            "resource": Object {
              "path": "/",
            },
          },
        ]
      }
    />
    <CollectionPreview
      description="baz"
      key="collection-2"
      link={null}
      resources={
        Array [
          Object {
            "id": "5",
            "resource": Object {
              "path": "/",
            },
          },
          Object {
            "id": "6",
            "resource": Object {
              "path": "/",
            },
          },
          Object {
            "id": "7",
            "resource": Object {
              "path": "/",
            },
          },
        ]
      }
    />
  </Main>
</Connect(Layout)>
`;
